test/ShallowWrapper-spec.jsx

```diff
@@ -2777,13 +2777,27 @@ describe('shallow', () => {
         ]);
       });
 
-      it('calls expected methods when receiving new context', () => {
-        wrapper.setContext({ foo: 'foo' });
-        expect(spy.args).to.deep.equal([
-          ['shouldComponentUpdate'],
-          ['componentWillUpdate'],
-          ['render'],
-        ]);
+      describeIf(REACT013 || REACT15, 'setContext', () => {
+        it('calls expected methods when receiving new context', () => {
+          wrapper.setContext({ foo: 'foo' });
+          expect(spy.args).to.deep.equal([
+            ['componentWillReceiveProps'],
+            ['shouldComponentUpdate'],
+            ['componentWillUpdate'],
+            ['render'],
+          ]);
+        });
+      });
+
+      describeIf(REACT014, 'setContext', () => {
```

Comment:
I'm glad we're testing this on all React versions - could you elaborate on why `componentWillReceiveProps` fires in 13 and 15, but not 14?